File: Leetcode/Divisor Game/Divisor Game.py
class Solution(object):
    def divisorGame(self, N):
        """
          We know that if N==1, we cannot choose any x.
         Also, we can always choose x=1 so that if answer for N is true,
         answer for N-1 will be false.
         Thus, Alice will always win if N is even number and if N is odd, she will
        always lose.
        """
        if(N%2==0):
            return True
        
        else:
            return False;
        

File: Leetcode/Divisor Game/Divisor Game.cpp
// This is the most optimized code for the following question, Time complexity for the code is O(1).
class Solution {
public:
    bool divisorGame(int N) {
        // This states that whenever the Value of N is even then it returns true else false
        return N%2==0;
    }
};
// Logic behind :-
// Basically since Alice will start playing so she will won at n=2 for sure,
//  then she will be  definitely wining at n=4 and 
// so on at all even no.s this thing happen. Similarly for odd no.s they are losing it.